service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -247,6 +254,24 @@ public C2DocumentBundle getLastC2DocumentBundle() {
             .orElse(null);
     }
 
+    @JsonIgnore
+    public C2DocumentBundle getC2DocumentBundleByUUID(UUID elementId) {
+        return c2DocumentBundle.stream()
+            .filter(c2DocumentBundleElement -> c2DocumentBundleElement.getId().equals(elementId))
+            .map(Element::getValue)
+            .findFirst()
+            .orElse(null);
+    }
+
+    public DynamicList buildC2DocumentDynamicList(UUID selected) {
+        AtomicInteger i = new AtomicInteger(1);
+        return asDynamicList(c2DocumentBundle, selected, documentBundle -> documentBundle.toLabel(i.getAndIncrement()));
```

Comment:
I would consider different approach than AtomicInteger, AtomicInteger is multithreaded construct and I am not sure if I would use it in that scenario. You can always consider int[1]

(Sorry for some reason I don't like the cannon like approach :D)